text/0032-dust-protection/0032-dust-protection.md

```diff
@@ -1,121 +1,1279 @@
+
+## Motivation
+
+In a distributed ledger network, every participant, a so called node, needs to keep track of the current ledger state. Since `chrysalis-pt2`, the IOTA ledger state is based on the UTXO model, where every node keeps track of all the currently unspent outputs. Without a `dust protection`, even outputs containing only one single IOTA coin are valid and therefore stored in the database.
+
+Misusage by honest users or intentionally bad behavior by malicious actors can lead to growing database and snapshot sizes and increasing computational costs (database lookups, balance calculations). Due to these increasing hardware requirements, the entry barrier to participate in the network becomes unaffordable and less nodes would operate the network.
+
+Especially in a fee-less system like IOTA, this is a serious issue, since an attacker can create a lot of damage with low effort. Other DLTs do not face this problem yet today, because such an attack would be much more expensive due to the high transaction fees.
+
+Solving the DLT scalability dilemma ultimatively leads to lower transaction costs. Therefore other DLT projects will also run into the same dust obstacle in the end. IOTA solves this problem with the `dust protection`.
+
+## Requirements
+
+- The maximum possible ledger database size must be limited to a reasonable and manageable size.
+- The `dust protection` must not depend on a global shared state of the ledger, so that transaction validation can happen in parallel.
+- The `dust protection` should work for outputs with arbitrary data and size.
+- The ledger database size should be fairly allocated to users based on the cryptograhically proven scarce resource, IOTA coins.
+
+## Detailed Design
+
+The current `dust protection` solution in `chrysalis-pt2` does not satisfy the mentioned requirements for the following reasons:
+
+- The enforced maximum limit of disc space is ~6.5 TB.
+- The dust allowance mechanism depends on total amount of funds in `DustAllowanceOutput` per address, which is a global shared state.
+- It is designed for one fixed output size.
+
+Therefore a new transaction validation rule is introduced which replaces the former `dust protection` solution completely.
+
+Messages including payloads, even transaction payloads, are considered to be pruned by the nodes, but unspent transaction outputs must be kept until they are spent. Therefore the `dust protection` is based on the unspent outputs only.
+
+**Every output created by a transaction needs to have at least a minimum amount of IOTA coins deposited in the output itself, otherwise the output is syntactically invalid.**
+
+min_deposit_of_output = v_byte_cost × v_byte<br>
+v_byte = ∑(weight<sub>n</sub> × field_byte_size<sub>n</sub>) + offset
```

Comment:
Currently the weight for the fields are arbitrarily set to 1.0 and 10.0. In the future these could become fractional values, so we should decide if we want to round min_deposit_of_output up or down and specify this in the RFC.

Rounding down has some advantages with regard to integer conversion.

Reply:
Added :+1: